src/Microsoft.Framework.ConfigurationModel/Configuration.cs

```diff
@@ -138,12 +138,15 @@ public IConfiguration GetSubKey(string key)
 
         public IConfigurationSourceRoot Add(IConfigurationSource configurationSource)
         {
-            configurationSource.Load();
-            return AddLoadedSource(configurationSource);
+            return Add(configurationSource, load: true);
         }
 
-        internal IConfigurationSourceRoot AddLoadedSource(IConfigurationSource configurationSource)
+        public IConfigurationSourceRoot Add(IConfigurationSource configurationSource, bool load)
```

Comment:
Sure. Btw, the `Add` overload with 2 arguments is just on this implementation. Most users will use configuration through `IConfiguration` which doesn't expose the overload